uniffi_bindgen/src/bindings/kotlin/gen_kotlin.rs

```diff
@@ -71,299 +72,62 @@ impl MergeWith for Config {
+        self
     }
 }
 
-#[derive(Default)]
-pub struct KotlinCodeOracle;
-
-impl KotlinCodeOracle {
-    fn create_code_type(&self, type_: TypeIdentifier) -> Box<dyn CodeType> {
-        // I really want access to the ComponentInterface here so I can look up the interface::{Enum, Record, Error, Object, etc}
-        // However, there's some violence and gore I need to do to (temporarily) make the oracle usable from filters.
-
-        // Some refactor of the templates is needed to make progress here: I think most of the filter functions need to take an &dyn CodeOracle
-        match type_ {
-            Type::UInt8 => Box::new(primitives::UInt8CodeType),
-            Type::Int8 => Box::new(primitives::Int8CodeType),
-            Type::UInt16 => Box::new(primitives::UInt16CodeType),
-            Type::Int16 => Box::new(primitives::Int16CodeType),
-            Type::UInt32 => Box::new(primitives::UInt32CodeType),
-            Type::Int32 => Box::new(primitives::Int32CodeType),
-            Type::UInt64 => Box::new(primitives::UInt64CodeType),
-            Type::Int64 => Box::new(primitives::Int64CodeType),
-            Type::Float32 => Box::new(primitives::Float32CodeType),
-            Type::Float64 => Box::new(primitives::Float64CodeType),
-            Type::Boolean => Box::new(primitives::BooleanCodeType),
-            Type::String => Box::new(primitives::StringCodeType),
-
-            Type::Timestamp => Box::new(miscellany::TimestampCodeType),
-            Type::Duration => Box::new(miscellany::DurationCodeType),
-
-            Type::Enum(id) => Box::new(enum_::EnumCodeType::new(id)),
-            Type::Object(id) => Box::new(object::ObjectCodeType::new(id)),
-            Type::Record(id) => Box::new(record::RecordCodeType::new(id)),
-            Type::Error(id) => Box::new(error::ErrorCodeType::new(id)),
-            Type::CallbackInterface(id) => {
-                Box::new(callback_interface::CallbackInterfaceCodeType::new(id))
-            }
-
-            Type::Optional(ref inner) => {
-                let outer = type_.clone();
-                let inner = *inner.to_owned();
-                Box::new(compounds::OptionalCodeType::new(inner, outer))
-            }
-            Type::Sequence(ref inner) => {
-                let outer = type_.clone();
-                let inner = *inner.to_owned();
-                Box::new(compounds::SequenceCodeType::new(inner, outer))
-            }
-            Type::Map(ref inner) => {
-                let outer = type_.clone();
-                let inner = *inner.to_owned();
-                Box::new(compounds::MapCodeType::new(inner, outer))
-            }
-            Type::External { .. } => panic!("no support for external types yet"),
-            Type::Wrapped { .. } => panic!("no support for wrapped types yet"),
-        }
-    }
+#[derive(Template)]
+#[template(syntax = "kt", escape = "none", path = "wrapper.kt")]
+pub struct KotlinWrapper<'a> {
```

Comment:
Given the size of the change you're making here, I suggest taking the opportunity to call this something other than "wrapper" and avoid any future confusion with "wrapper types". `KotlinBindings` maybe?